File: beginners/1012/main.go
package main

import (
	"fmt"
)

type shape interface {
	area() float64
	getName() string
}

type triangle struct {
	Name         string
	base, height float64
}

func (s *triangle) area() float64 {
	return s.base * s.height / 2.0
}

func (s *triangle) getName() string {
	return s.Name
}

type circle struct {
	Name   string
	radius float64
}

func (s *circle) area() float64 {
	return s.radius * s.radius * 3.14159
}

func (s *circle) getName() string {
	return s.Name
}

type trapezoid struct {
	Name                 string
	base1, base2, height float64
}

func (s *trapezoid) area() float64 {
	return ((s.base1 + s.base2) * s.height) / 2
}

func (s *trapezoid) getName() string {
	return s.Name
}

type square struct {
	Name string
	side float64
}

func (s *square) area() float64 {
	return s.side * s.side
}

func (s *square) getName() string {
	return s.Name
}

type rectangle struct {
	Name          string
	width, height float64
}

func (s *rectangle) area() float64 {
	return s.width * s.height
}

func (s *rectangle) getName() string {
	return s.Name
}

func print(shapes []shape) {
	for _, s := range shapes {
		fmt.Printf("%s: %.3f\n", s.getName(), s.area())
	}
}

func main() {

	var a, b, c float64
	fmt.Scanf("%f %f %f", &a, &b, &c)

	shapes := []shape{
		&triangle{Name: "TRIANGULO", base: a, height: c},
		&circle{Name: "CIRCULO", radius: c},
		&trapezoid{Name: "TRAPEZIO", base1: a, base2: b, height: c},
		&square{Name: "QUADRADO", side: b},
		&rectangle{Name: "RETANGULO", width: a, height: b},
	}

	print(shapes)
}

File: beginners/1012/main.py
a, b, c = input().split(" ")
a, b, c = float(a), float(b), float(c)

class shape:
    height = 0
    width = 0
    
    def __init__(self, height = 0, width = 0):
        self.height = height
        self.width = width
    
    def __str__(self) -> str:
        return f"{self.__class__.__name__.upper()}: {self.area:.3f}"
    
    def calculate_area():
        pass

class triangulo(shape):
    def calculate_area(self):
        self.area = (self.height * self.width) / 2
    
class circulo(shape):
    def calculate_area(self):
        self.area = 3.14159 * self.height**2

class trapezio(shape):
    def calculate_area(self):
        self.area = (self.height * self.width) / 2

class quadrado(shape):
    def calculate_area(self):
        self.area = self.height ** 2

class retangulo(shape):
    def calculate_area(self):
        self.area = self.height * self.width

for s in [triangulo(a, c), circulo(c), trapezio((a+b), c), quadrado(b, b), retangulo(a, b)]:
    s.calculate_area()
    print(s)
